### PR TITLE
[#3783] improvement(test): add test 'list-all-tables' for the Hive catalog

### PR DESCRIPTION
**Pull request by chenyuan99**
### What changes were proposed in this pull request?

add test 'list-all-tables'  for the Hive catalog

### Why are the changes needed?

Fix: #3783 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

 CI passed

### CODE REVIEW

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/CatalogHiveIT.java [hunk 1]
```diff
@@ -629,6 +630,9 @@ public void testListTables() {
     catalog
         .asTableCatalog()
         .createTable(hudiTable, createColumns(), null, ImmutableMap.of("provider", "hudi"));
+    catalog
+        .asTableCatalog()
+        .createTable(hiveTable, createColumns(), null, ImmutableMap.of("provider", "hive"));
     NameIdentifier[] tables = catalog.asTableCatalog().listTables(Namespace.of(schemaName));
     Assertions.assertEquals(0, tables.length);
```
Comment (mchades) [inline catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/CatalogHiveIT.java:637]: The expected table number needs to be updated to 1

**Comment by mchades** [comment]
plz update the PR description to follow the template

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/CatalogHiveIT.java (hunk 1 above)
Comment (chenyuan99) [reply catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/CatalogHiveIT.java]: I see.

**Review by mchades — APPROVED** [review]
LGTM. Thanks for your contributions!

**Pull request merged**